[20.09] tailscale: 1.4.4 -> 1.8.3

blitz opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

This MR backports tailscale updates to the 20.09 release. Tailscale is updated to 1.8.3, which is the latest stable tailscale release. See https://pkgs.tailscale.com/stable/#static

This backports #118742, #116591, #114420,  #121946, #122631.

I've tested v1.8.0  locally and the VPN plus DNS works.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

kalbasit committed:
tailscale: 1.4.4 -> 1.4.5 (#114420)

(cherry picked from commit d470cd98c12358c1e5af00011e9c247054c17d9b)

LeSuisse committed:
tailscale: 1.4.5 -> 1.6.0

(cherry picked from commit ce403df042d7468d338d0019219bb1c224b7930d)

blitz commented:
`nox-review` is happy:

```
nox-review pr 121400
=== Reviewing PR 121400 : [20.09] tailscale: 1.4.4 -> 1.6.0
==> Creating nixpkgs repo in /home/julian/.nox/nixpkgs
==> We're in a git repo, trying to fetch it
==> Fetching base (release-20.09)
==> Fetching PR
==> Fetching extra history for merging
==> Merging PR into base
Building in /run/user/1000/nox-review-9pg57bxl: tailscale
Invoking nix-build -A tailscale /home/julian/.nox/nixpkgs
/nix/store/2crmw2rkjwdnwckbk0dvb49l46jdpjkh-tailscale-1.6.0
Result in /run/user/1000/nox-review-9pg57bxl
total 0
lrwxrwxrwx 1 julian users 59 May  1 16:56 result -> /nix/store/2crmw2rkjwdnwckbk0dvb49l46jdpjkh-tailscale-1.6.0
```

FRidh commented:
I think af3ecb1 should not be backported; it will results in problem when another tool is used for managing the network. Therefore, I am still considering reverting it on master.

danderson commented:
Hmm, good point. That behavior is fixed in 1.8, and 1.8 is releasing soon. So I suggest either merging this without the openresolv dep, or waiting for 1.8 and backporting that to get good DNS behavior.

blitz commented:
@danderson 

> So I suggest either merging this without the openresolv dep, or waiting for 1.8 and backporting that to get good DNS behavior.

I've dropped the openresolv dependency and retested. Looks good as well.

blitz committed:
tailscale: 1.6.0 -> 1.8.0

(cherry picked from commit 5686421c0ae8d4ec8f10f7d4192cc6eef5d3e665)

blitz commented:
Mmh. Backporting the 1.8 version fails to build:

```
Building subPackage ./cmd/tailscale
cmd/tailscale/cli/web.go:10:2: cannot find package "." in:
        /build/source/vendor/embed
error: builder for '/nix/store/cpfnky7vnj4sil9bx9c631mlki65xz07-tailscale-1.8.0.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/62b3wvqwxb0f3bvxfnn1pnri7nki4n3p-source
       > source root is source
       > setting SOURCE_DATE_EPOCH to timestamp 1620323390 of file source/wgengine/winnet/winnet_windows.go
       > patching sources
       > configuring
       > building
       > Building subPackage ./cmd/tailscale
       > cmd/tailscale/cli/web.go:10:2: cannot find package "." in:
       >   /build/source/vendor/embed
       For full logs, run 'nix log /nix/store/cpfnky7vnj4sil9bx9c631mlki65xz07-tailscale-1.8.0.drv'.

```

danderson commented:
Tailscale 1.8 requires Go 1.16, which I suspect isn't backported to 20.09 - or at least is not whatever we pick as the build dependency.

danderson commented:
Also, I'm about to cut 1.8.1 with a couple linux-specific fixes, so you may want to hold off slightly for that :)

(none of the bugs should affect NixOS, it's compat with older Ubuntus, but still, freshness etc.)

kalbasit commented:
> Tailscale 1.8 requires Go 1.16, which I suspect isn't backported to 20.09 - or at least is not whatever we pick as the build dependency.

We did backport Go 1.16 to release 20.09 you just have to pin tailscale to use it. https://github.com/NixOS/nixpkgs/blob/release-20.09/pkgs/development/compilers/go/1.16.nix

blitz committed:
tailscale: build with Go 1.16

Tailscale 1.8 requires a newer Go than what is currently the default
in NixOS 20.09.

blitz commented:
> > Tailscale 1.8 requires Go 1.16, which I suspect isn't backported to 20.09 - or at least is not whatever we pick as the build dependency.
> 
> We did backport Go 1.16 to release 20.09 you just have to pin tailscale to use it. https://github.com/NixOS/nixpkgs/blob/release-20.09/pkgs/development/compilers/go/1.16.nix

Thanks. Building it with Go 1.16 indeed fixes the build. I'm giving this a spin on my stable box right now.

blitz commented:
> I'm giving this a spin on my stable box right now.

I see no issues so far. But we might want to wait for 1.8.1 as @danderson suggested.

danderson commented:
1.8.1 actually introduced a different regression that might affect some NixOS users. I would suggest either proceeding with 1.8.0 (the bugs in there don't matter for NixOS), or waiting for 1.8.2 (probably tomorrow).

LeSuisse committed:
tailscale: 1.8.0 -> 1.8.3

(cherry picked from commit a3c9fd29fe88fb9e1dbf34513907cf5bb991cbca)

blitz commented:
@danderson I've rebased and backported 1.8.3 as well.

blitz commented:
Closing this, as there is probably little interest to merge this into 20.09 since 21.05 is out.
